### PR TITLE
fixed potential NPE

### PR DESCRIPTION
**Pull request by kaikreuzer**
Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=469378
Signed-off-by: Kai Kreuzer kai@openhab.org

### CODE REVIEW

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java [hunk 1]
```diff
@@ -287,20 +287,17 @@ public String getLabel(Widget w) {
...
+                if (getFormatPattern(label)==null && item.getStateDescription() != null
                         && item.getStateDescription().getPattern() != null) {
                     label = label + " [" + item.getStateDescription().getPattern() + "]";
                 }
 
-                if (label.contains("[")) {
-                    indexOpenBracket = label.indexOf("[");
-                    indexCloseBracket = label.indexOf("]");
-                    formatPattern = label.substring(indexOpenBracket + 1, indexCloseBracket);
+                String updatedPattern = getFormatPattern(label);
```
Comment (maggu2810) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: We could move this a line before the if statement and use the variable in the first test.
So no need to call the function twice. Sure, no problem at all, but I try to reduce the function calls if possible.

Comment (maggu2810) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: Okay, forgett my previous comment, i deleted it. ;-)

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java [hunk 2]
```diff
@@ -287,20 +287,17 @@ public String getLabel(Widget w) {
         if (itemName != null) {
             State state = null;
             String formatPattern = null;
-            int indexOpenBracket = label.indexOf("[");
-            int indexCloseBracket = label.indexOf("]");
 
             try {
                 Item item = getItem(itemName);
-                if (!label.contains("[") && item.getStateDescription() != null
+                if (getFormatPattern(label)==null && item.getStateDescription() != null
```
Comment (maggu2810) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: Shouldn't we use spaces in front and after ==?

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java [hunk 3]
```diff
@@ -392,7 +401,7 @@ protected String formatUndefined(String formatPattern) {
      * (e.g. "[MAP(en.map):%s]") and execute the transformation in this case.
      */
     private String transform(String label) {
-        if (label.contains("[") && label.endsWith("]")) {
+        if (getFormatPattern(label)!=null) {
```
Comment (maggu2810) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: Shouldn't we use spaces in front and after ==?

**Comment by kaikreuzer** [comment]
@maggu2810 you are of course right, I forgot to apply the formatter. Should be fixed now.

**Comment by maggu2810** [comment]
Cannot test this at the moment, but the code changes looks good for me.

**Pull request merged**